esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -1985,47 +1987,8 @@ void WsWuInfo::getWorkunitThorLog(const char* fileName, MemoryBuffer& buf, const
     OwnedIFileIO rIO = rFile->openShared(IFOread,IFSHfull);
     if (!rIO)
         throw MakeStringException(ECLWATCH_CANNOT_READ_FILE,"Cannot read file %s.",fileName);
-    OwnedIFileIOStream ios = createBufferedIOStream(rIO);
-
-    StringBuffer line;
-    bool eof = false;
-    bool include = false;
-
-    VStringBuffer startwuid("Started wuid=%s", wuid.str());
-    VStringBuffer endwuid("Finished wuid=%s", wuid.str());
-
-    const char *sw = startwuid.str();
-    const char *ew = endwuid.str();
 
-    Owned<IFileIOStream> outIOS;
-    if (!isEmptyString(outFile))
-    {
-        CWsWuFileHelper helper(nullptr);
-        outIOS.setown(helper.createIOStreamWithFileName(outFile, IFOcreate));
-    }
-    while (!eof)
-    {
-        line.clear();
-        for (;;)
-        {
-            char c;
-            size32_t numRead = ios->read(1, &c);
-            if (!numRead)
-            {
-                eof = true;
-                break;
-            }
-            line.append(c);
-            if (c=='\n')
-                break;
-        }
-        if (strstr(line.str(), sw))
-            include = true;
-        if (include)
-            outputALine(line.length(), line.str(), buf, outIOS);
-        if (strstr(line.str(), ew))
-            include = false;
-    }
+    readWorkunitLog(createBufferedIOStream(rIO), buf, outFile);
```

Comment:
This is going to leak...
Because readWorkunitLog(OwnedIFileIOStream ios, is going to link/release itself.
Nothing is owns the object being created here, so it will be leaked.

See other comments re. readWorkunitLog(OwnedIFileIOStream, issue.

Reply:
Will fix